components/feature/prompts/src/main/java/mozilla/components/feature/prompts/ColorPickerDialogFragment.kt

```diff
@@ -35,137 +33,91 @@ internal class ColorPickerDialogFragment : PromptDialogFragment() {
-            val arguments = fragment.arguments ?: Bundle()
-
-            with(arguments) {
-                putString(KEY_SESSION_ID, sessionId)
-                putInt(KEY_SELECTED_COLOR, defaultColor.toColor())
-            }
-
-            fragment.arguments = arguments
-
-            return fragment
+    override fun onClick(dialog: DialogInterface?, which: Int) {
+        when (which) {
+            DialogInterface.BUTTON_POSITIVE -> feature?.onConfirm(sessionId, selectedColor.toHexColor())
+            DialogInterface.BUTTON_NEGATIVE -> feature?.onCancel(sessionId)
         }
     }
 
     @SuppressLint("InflateParams")
     internal fun createDialogContentView(): View {
-        val inflater = LayoutInflater.from(requireContext())
-        val view = inflater.inflate(R.layout.mozac_feature_prompts_color_picker_dialogs, null)
-
-        initSelectedColor(view)
-        initRecyclerView(view, inflater)
-
-        return view
-    }
-
-    private fun onPositiveClickAction() {
-        feature?.onConfirm(sessionId, selectedColor.toHexColor())
-    }
-
-    fun onColorChange(newColor: Int) {
-        selectedColor = newColor
-        selectColorTexView.changeColor(newColor)
-    }
+        val view = LayoutInflater
+            .from(requireContext())
+            .inflate(R.layout.mozac_feature_prompts_color_picker_dialogs, null)
+
+        // Load list of colors from resources
+        val typedArray = resources.obtainTypedArray(R.array.mozac_feature_prompts_default_colors)
+        defaultColors = List(typedArray.length()) { i ->
+            val color = typedArray.getColor(i, Color.BLACK)
+            ColorItem(color = color, contentDescription = color.toHexColor())
+        }
+        typedArray.recycle()
 
-    /**
-     * RecyclerView adapter for displaying color items.
-     */
-    internal class ColorAdapter(
-        private val fragment: ColorPickerDialogFragment,
-        private val inflater: LayoutInflater
-    ) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
-
-        @Suppress("MagicNumber")
-        internal val defaultColors = arrayOf(
-            Color.rgb(215, 57, 32),
-            Color.rgb(255, 134, 5),
-            Color.rgb(255, 203, 19),
-            Color.rgb(95, 173, 71),
-            Color.rgb(33, 161, 222),
-            Color.rgb(16, 36, 87),
-            Color.rgb(91, 32, 103),
-            Color.rgb(212, 221, 228),
-            Color.WHITE
-        )
-
-        override fun getItemCount(): Int = defaultColors.size
-
-        override fun onCreateViewHolder(parent: ViewGroup, type: Int): RecyclerView.ViewHolder {
-            val view = inflater.inflate(R.layout.mozac_feature_prompts_color_item, parent, false)
-            return ColorViewHolder(view)
+        // Set up recycler view
+        listAdapter = BasicColorAdapter(this::onColorChange)
+        view.findViewById<RecyclerView>(R.id.recyclerView).apply {
+            layoutManager = LinearLayoutManager(context, RecyclerView.VERTICAL, false)
```

Comment:
The initial selected color is not visible as is the last element of the list, something that not happens on Fennec. You could test with this [site](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/color).

![Screenshot_20190628-120530](https://user-images.githubusercontent.com/773158/60356359-c3c71f00-999e-11e9-886a-e0b1b8f22833.png)

I think could add `stackFromEnd = true` to the `layoutManager` that will fix that.